src/libstd/sync/once.rs

```diff
@@ -114,15 +114,15 @@ impl Once {
         // otherwise we run the job and record how many people will try to grab
         // this lock
         let guard = self.mutex.lock();
-        if self.cnt.load(Ordering::SeqCst) > 0 {
+        if self.cnt.load(Ordering::Acquire) > 0 {
             f();
-            let prev = self.cnt.swap(isize::MIN, Ordering::SeqCst);
-            self.lock_cnt.store(prev, Ordering::SeqCst);
+            let prev = self.cnt.swap(isize::MIN, Ordering::AcqRel);
+            self.lock_cnt.store(prev, Ordering::Release);
         }
         drop(guard);
 
         // Last one out cleans up after everyone else, no leaks!
-        if self.lock_cnt.fetch_add(-1, Ordering::SeqCst) == 1 {
+        if self.lock_cnt.fetch_add(-1, Ordering::AcqRel) == 1 {
```

Comment:
This can be `Relaxed`: the necessary synchronization to know that the lock is ready to be cleaned up is provided by the lock itself.